### DISCUSSION TITLE
CVEs in docker image and Rust SDK

### DISCUSSION THREAD

**Discussion by yoav-lavi** [zitadel/zitadel#8428] [Q&A] [answered]
Hi, I noticed there were a few open CVEs in the Docker image and the Rust SDK: 

[CVE-2024-29891⁠](https://scout.docker.com/v/CVE-2024-29891?s=golang&n=zitadel&ns=github.com%2Fzitadel&t=golang&vr=%3E%3D0&utm_source=desktop&utm_medium=ExternalLink)
[CVE-2024-28197⁠](https://scout.docker.com/v/CVE-2024-28197?s=golang&n=zitadel&ns=github.com%2Fzitadel&t=golang&vr=%3E%3D0&utm_source=desktop&utm_medium=ExternalLink)
[CVE-2024-41953](https://scout.docker.com/v/CVE-2024-41953?s=golang&n=zitadel&ns=github.com%2Fzitadel&t=golang&vr=%3E%3D0&utm_source=desktop&utm_medium=ExternalLink)
[CVE-2024-41952⁠](https://scout.docker.com/v/CVE-2024-41952?s=golang&n=zitadel&ns=github.com%2Fzitadel&t=golang&vr=%3E%3D0&utm_source=desktop&utm_medium=ExternalLink)
[CVE-2024-39683⁠](https://scout.docker.com/v/CVE-2024-39683?s=golang&n=zitadel&ns=github.com%2Fzitadel&t=golang&vr=%3E%3D0&utm_source=desktop&utm_medium=ExternalLink)
[CVE-2024-32967⁠](https://scout.docker.com/v/CVE-2024-32967?s=golang&n=zitadel&ns=github.com%2Fzitadel&t=golang&vr=%3E%3D0&utm_source=desktop&utm_medium=ExternalLink)
[CVE-2024-32868⁠](https://scout.docker.com/v/CVE-2024-32868?s=golang&n=zitadel&ns=github.com%2Fzitadel&t=golang&vr=%3E%3D0&utm_source=desktop&utm_medium=ExternalLink)
[CVE-2024-29892⁠](https://scout.docker.com/v/CVE-2024-29892?s=golang&n=zitadel&ns=github.com%2Fzitadel&t=golang&vr=%3E%3D0&utm_source=desktop&utm_medium=ExternalLink)
[RUSTSEC-2023-0071](https://github.com/smartive/zitadel-rust/issues/512)

Was wondering if these were being addressed / have been addressed and how to proceed in regards to them. 

Thanks!

**Comment by fforootd** [comment, marked as answer]
Hi, I am not fully sure what you mean by "a few open CVEs in the Docker image".

As we publish CVE's (and these look like ours) we also tend to publish updated versions. 
You can find the advisories here with the according patches https://github.com/zitadel/zitadel/security/advisories

The rust library is maintained by another party but we can try and nudge someone there.

Hope this helps

**Reply by yoav-lavi** [reply]
Right, thanks, I was going by the automated Docker vulnerability scanner and just now saw that these were patched, not sure why the image still lists them. Thanks!

**Reply by fforootd** [reply]
Haha no worries, good to know that we did not miss something 😁 

Let me investigate why docker scout did not parse the fixed versions 🫨
